template/src/components/app.tsx

```diff
@@ -1,23 +1,21 @@
-import { FunctionalComponent, h } from 'preact';
+import { h } from 'preact';
 import { Route, Router } from 'preact-router';
 
+import Header from './header';
+
+// Code-splitting is automated for `routes` directory
 import Home from '../routes/home';
 import Profile from '../routes/profile';
-import NotFoundPage from '../routes/notfound';
-import Header from './header';
 
-const App: FunctionalComponent = () => {
-    return (
-        <div id="preact_root">
-            <Header />
-            <Router>
-                <Route path="/" component={Home} />
-                <Route path="/profile/" component={Profile} user="me" />
-                <Route path="/profile/:user" component={Profile} />
-                <NotFoundPage default />
-            </Router>
-        </div>
-    );
-};
+const App = () => (
+	<div id="app">
+        <Header />
+        <Router>
+            <Route path="/" component={Home} />
```

Comment:
We still use `<Route>` while the default template does not, as this is the type-safe API. Appending props to components to use them as routes will (rightfully) trigger TS errors.